examples/tv-app/android/java/MessagesManager.cpp

```diff
@@ -0,0 +1,425 @@
+
+        jint jcontrol  = static_cast<jint>(messageControl.Raw());
+        jint jduration = -1;
+        if (duration.IsNull())
+        {
+            jduration = static_cast<jint>(duration.Value());
+        }
+        jlong jstartTime = -1;
+        if (startTime.IsNull())
+        {
+            jstartTime = static_cast<jlong>(startTime.Value());
+        }
+
+        jint jniPriority = static_cast<jint>(priority);
+        jclass priorityTypeClass;
+        VerifyOrReturnError(CHIP_NO_ERROR !=
+                                chip::JniReferences::GetInstance().GetLocalClassRef(
+                                    env, "com/matter/tv/server/tvapp/Message$PriorityType", priorityTypeClass),
+                            CHIP_ERROR_INCORRECT_STATE, ChipLogError(Zcl, "Could not find class Message$PriorityType"));
+        jmethodID priorityTypeCtor = env->GetMethodID(priorityTypeClass, "<init>", "(I)V");
+        VerifyOrReturnError(priorityTypeCtor != nullptr, CHIP_ERROR_INCORRECT_STATE,
+                            ChipLogError(Zcl, "Could not find Message$PriorityType constructor"));
+        jobject jpriority = env->NewObject(priorityTypeClass, priorityTypeCtor, jniPriority);
+
+        jclass vectorClass = env->FindClass("java/util/Vector");
+        VerifyOrReturnError(vectorClass != nullptr, CHIP_ERROR_INCORRECT_STATE, ChipLogError(Zcl, "Could not find class Vector"));
+        jmethodID vectorCtor = env->GetMethodID(priorityTypeClass, "<init>", "()V");
+        VerifyOrReturnError(vectorCtor != nullptr, CHIP_ERROR_INCORRECT_STATE,
+                            ChipLogError(Zcl, "Could not find Vector constructor"));
+        jobject joptions = env->NewObject(vectorClass, vectorCtor);
+
+        jmethodID vectorAdd = env->GetMethodID(priorityTypeClass, "add", "(Ljava/lang/Object;)Z");
+
+        if (responses.HasValue())
+        {
+            jclass messageResponseOptionClass;
+            VerifyOrReturnError(CHIP_NO_ERROR !=
+                                    chip::JniReferences::GetInstance().GetLocalClassRef(
+                                        env, "com/matter/tv/server/tvapp/MessageResponseOption", messageResponseOptionClass),
+                                CHIP_ERROR_INCORRECT_STATE, ChipLogError(Zcl, "Could not find class MessageResponseOption"));
+            jmethodID messageResponseOptionCtor = env->GetMethodID(messageResponseOptionClass, "<init>", "(JLjava/lang/String;)V");
+            VerifyOrReturnError(messageResponseOptionCtor != nullptr, CHIP_ERROR_INCORRECT_STATE,
+                                ChipLogError(Zcl, "Could not find MessageResponseOption constructor"));
+
+            auto iter = responses.Value().begin();
+            while (iter.Next())
+            {
+                auto & response = iter.GetValue();
+
+                jlong jniid = static_cast<jlong>(response.messageResponseID.Value());
+                UtfString jlabel(env, response.label.Value());
```

Comment:
it seems UtfString is redundant, where UtfString create globalRef, and release it when it is out of scope in this function, it is better to use mEnv->NewStringUTF(hex_buf). when the data is passed from jni to java, Java would take care of the lifetime and decide when to release it.

Reply:
addressed